### PR TITLE
fix: move environment checks to a dedicated monitor

### PR DESCRIPTION
**Pull request by derekbit**
#### Which issue(s) this PR fixes:

Issue longhorn/longhorn#9983

#### What this PR does / why we need it:

- Move environment checks to a dedicated monitor
- Remove hugepage condition after disabling v2 data engine
- Increase the environment check period to 30 minutes

### CODE REVIEW

**Review by c3y1huang — DISMISSED** [review]
LGTM

File: controller/node_controller.go [hunk 1]
```diff
@@ -1834,6 +1467,16 @@ func (nc *NodeController) syncWithDiskMonitor(node *longhorn.Node) (map[string]*
...
+func (nc *NodeController) syncWithEnvironmentCheckMonitor() ([]longhorn.Condition, error) {
+	v, err := nc.environmentCheckMonitor.GetCollectedData()
+	if err != nil {
+		return []longhorn.Condition{}, err
+	}
+
+	conditions := v.([]longhorn.Condition)
+	return conditions, nil
+}
+
```
[the remark this answers is not in the document]
Comment (derekbit) [reply controller/node_controller.go]: Resolved. Thanks.

File: controller/monitor/environment_check_monitor.go [hunk 2]
```diff
@@ -0,0 +1,558 @@
...
+	case strings.Contains(osImage, "centos"):
+		fallthrough
+	case strings.Contains(osImage, "fedora"):
+		fallthrough
+	case strings.Contains(osImage, "red hat"):
+		fallthrough
+	case strings.Contains(osImage, "rocky"):
+		fallthrough
+	case strings.Contains(osImage, "ol"):
+		queryPackagesCmd = "rpm"
```
Comment (mantissahz) [inline controller/monitor/environment_check_monitor.go:202]: Do we still use `rpm` command  to check if pkgs are installed?

**Review by c3y1huang — APPROVED** [review]

File: controller/monitor/environment_check_monitor.go (hunk 2 above)
Comment (derekbit) [reply controller/monitor/environment_check_monitor.go:202]: The purpose of the PR is not to remove `rpm`. It just moves the checks to a monitor and prevents frequent checks. Also fix some bugs.

**Review by mantissahz — APPROVED** [review]
LGTM

**Comment by derekbit** [comment]
@mergify backport v1.8.x

**Comment by derekbit** [comment]
@mergify backport v1.7.x

**Pull request merged**